Assets/Scripts/Cards/DeckManager.cs

```diff
@@ -0,0 +1,448 @@
+// +-----------------------------------------------------------------+
+// @author - Ryan Herwig
+// @Contributers - 
+// @Last Modifieed - September 4 2024
+// @Description - A script with universal functions for a Generic deck
+// +-----------------------------------------------------------------+
+using System.Collections.Generic;
+using Debug = UnityEngine.Debug;
+using Random = UnityEngine.Random;
+
+//Generic Class
+// ** IMPORTANT ** - Does not inherit from MonoBehavior
+public class DeckManager<T>
+{
+    //Making Singleton Class
+    #region Singleton
+    private static DeckManager<T> instance;
+    public static DeckManager<T> Instance
+    {
+        get
+        {
+            if (instance == null)
+                instance = new DeckManager<T>();
+            return instance;
+        }
+        set
+        {
+            instance = value;
+        }
+    }
+    #endregion
+
+    #region Shuffle
+
+    /// <summary>
+    /// Shuffles the deck
+    /// </summary>
+    /// <param name="deck">the deck to be shuffled</param>
+    /// <returns>the shuffled deck</returns>
+    public List<T> Shuffle(List<T> deck)
+    {
+        //Checks if the deck is valid
+        if (deck == null || deck.Count < 1)
+            return deck;
+
+        int deckSize = deck.Count; //Size of the deck
```

Comment:
local variables do not get underscores so this script is good variable-naming wise